data-prepper-plugins/map-values-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/mapvalues/MapValues.java

```diff
@@ -0,0 +1,179 @@
+public class MapValues extends AbstractProcessor<Record<Event>, Record<Event>> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MapValues.class);
+    private final ExpressionEvaluator expressionEvaluator;
+    private final MapValuesConfig config;
+    private final LinkedHashMap<Range<Integer>, String> RANGES;
+    private final Map<String, String> MAP;
+
+    private final HashMap<String, String> PATTERNS;
+    @DataPrepperPluginConstructor
+    public MapValues(PluginMetrics pluginMetrics, final MapValuesConfig mapValuesConfig, final ExpressionEvaluator expressionEvaluator) {
+        super(pluginMetrics);
+        this.config = mapValuesConfig;
+        this.expressionEvaluator = expressionEvaluator;
+        // LOG.info("Entered mapvalues constructor");
+        MAP = new HashMap<>();
+        RANGES = new LinkedHashMap<>();
+        PATTERNS = config.getPatterns();
+        processMapField(mapValuesConfig.getMap());
+        parseFile(mapValuesConfig.getFilePath());
+
+    }
+
+    private void processMapField(Map<String, String> map){
+        for(Map.Entry<String, String> mapEntry : map.entrySet()){
+            String key = mapEntry.getKey();
+            if(key.contains(",")){
+                parseKeys(mapEntry);
+            }
+            else if(key.contains("-")){
+                parseRanges(mapEntry);
+            }
+            else{
+                MAP.put(key, map.get(key));
+            }
+        }
+    }
+
+    private void parseKeys(Map.Entry<String, String>  mapEntry){
+        String commaRegex = "([^,]+)(?:,|$)";
+        Pattern pattern = Pattern.compile(commaRegex);
+        Matcher matcher = pattern.matcher(mapEntry.getKey());
+        while (matcher.find()) {
+            String individualKey = matcher.group(1);
+            MAP.put(individualKey, mapEntry.getValue());
+        }
+    }
+    private void parseRanges(Map.Entry<String, String>  mapEntry){
+        try{
+            String rangeRegex = "^(\\d+)-(\\d+)$";
+            Pattern pattern = Pattern.compile(rangeRegex);
+            Matcher matcher = pattern.matcher(mapEntry.getKey());
+            if (matcher.find()) {
+                Integer low = Integer.parseInt(matcher.group(1));
+                Integer high = Integer.parseInt(matcher.group(2));
+                Range<Integer> rangeEntry = Range.between(low,high);
+                if(isOverlapping(rangeEntry)){
+                    LOG.error("map option contains a range [{}] which overlaps with other range entries.",
+                            mapEntry.getKey());
+                }
+                else{
+                    RANGES.put(Range.between(low,high), mapEntry.getValue());
+                }
+            }
+        }
+        catch (NumberFormatException ex){
+            LOG.error("map option contains a range which is not allowed : [{}].",
+                    mapEntry.getKey(), ex);
+        }
+    }
+
+    private boolean isOverlapping(Range<Integer> rangeEntry){
```

Comment:
I think we also want to check that the entries in the `MAP` are also not present in the ranges.

Reply:
Yeah, I need to add that check.